t/06elements-port.t

```diff
@@ -0,0 +1,555 @@
+    is 
+        @cn.elems, 2, 
+        'element has correct number of children after elem normalization';
+
+    # TEST
+    #ok ! $t2.parentNode.defined, '2nd text node has no parent';
+    # TEST
+    #ok ! $t3.parentNode.defined, '3rd text node has no parent';
+}
+
+# 4.2 Normalization on a Document node
+{
+    my $doc = XML::LibXML::Document.new();
+    my $t1 = $doc.createTextNode( "bar1" );
+    my $t2 = $doc.createTextNode( "bar2" );
+    my $t3 = $doc.createTextNode( "bar3" );
+    my $e  = $doc.createElement("foo");
+    my $e2 = $doc.createElement("bar");
+    $doc.setDocumentElement($e);
+    $e.appendChild( $e2 );
+    $e.appendChild( $t1 );
+    $e.appendChild( $t2 );
+    $e.appendChild( $t3 );
+
+    # this is the correct behaviour for DOM. the nodes are still
+    # referred
+    # TEST
+    #is 
+    #    $e.childNodes.elems, 4, 
+    #    'element has corrent number of children before normalization';
+    #$doc.normalize;
+
+    # TEST
+    is
+         $e.childNodes.elems, 2, 
+         'element has correct number of children after doc normalization';
+
+    # TEST
+    #ok ! $t2.parentNode.defined, ' TODO : Add test name');
+    # TEST
+    #ok ! $t3.parentNode.defined, ' TODO : Add test name');
+}
+
+# 5. XML::LibXML extensions
+{
+    my $plainstring = "foo";
+    my $stdentstring= "$foo \& this";
+
+    my $doc = XML::LibXML::Document.new();
+    my $elem = $doc.createElement( $foo );
+    $doc.setDocumentElement( $elem );
+
+    $elem.appendText( $plainstring );
+    # TEST
+    is
+        $elem.string_value , $plainstring, 
+        'element has correct string_value';
+
+    $elem.appendText( $stdentstring );
+    # TEST
+    is 
+        $elem.string_value, $plainstring ~ $stdentstring, 
+        'text was properly appended to element text';
+
+    $elem.appendTextChild( "foo" );
+    $elem.appendTextChild( "foo" => "foo\&bar" );
+
+    my @cn = $elem.childNodes;
+    # TEST
+    #ok( scalar(@cn), ' TODO : Add test name' );
+    # TEST
+    is 
+        @cn.elems, 3, 
+        'element has correct number of children';
+    # TEST
+    ok  ! @cn[1].hasChildNodes, 'second child has no descendants';
+    # TEST
+    # cw: Missing text node as descendant.
+    ok  @cn[2].hasChildNodes, 'third child has descendants';
+    # TEST (new) - See if two methods of counting children agree.
+    # cw: -XXX- Is this a fair test? Why does it fail?
+    ok  
+        @cn[2].childNodes.elems == @cn[2].elems,
+        'child count between perl6 and libxml2 agree';
+}
+
+# 6. XML::LibXML::Attr nodes
+{
+    my $dtd = q:to"EOF";
+<!DOCTYPE root [
+<!ELEMENT root EMPTY>
+<!ATTLIST root fixed CDATA  #FIXED "foo">
+<!ATTLIST root a:ns_fixed CDATA  #FIXED "ns_foo">
+<!ATTLIST root name NMTOKEN #IMPLIED>
+<!ENTITY ent "ENT">
+]>
+EOF
+
+    my $ns = 'urn:xx';
+    my $xml_nons = '<root foo="&quot;bar&ent;&quot;" xmlns:a="$ns"/>';
+    my $xml_ns = '<root xmlns="$ns" xmlns:a="$ns" foo="&quot;bar&ent;&quot;"/>';
+
+    # TEST:$xml=2;
```

Comment:
Tobias,

This is where things start to go off of the rails. Particularly when it comes to entity replacement.